11-payment-encoding.md

```diff
@@ -175,13 +182,21 @@ A writer:
         `fee_base_msat`, `fee_proportional_millionths`, and `cltv_expiry_delta` are as
         specified in [BOLT #7](07-routing-gossip.md#the-channel_update-message).
     - MAY include more than one `r` field to provide multiple routing options.
+  - if `9` contains non-zero bits:
+    - SHOULD use the minimum `data_length` possible.
+  - otherwise:
+    - MUST omit the `9` field altogether.
   - MUST pad field data to a multiple of 5 bits, using 0s.
   - if a writer offers more than one of any field type, it:
     - MUST specify the most-preferred field first, followed by less-preferred fields, in order.
 
 A reader:
   - MUST skip over unknown fields, OR an `f` field with unknown `version`, OR  `p`, `h`, or
   `n` fields that do NOT have `data_length`s of 52, 52, or 53, respectively.
+  - if the `9` field contains unknown _odd_ bits that are non-zero:
+    - MUST ignore the bit.
+  - if the `9` field contains unknown _even_ bits that are non-zero:
+    - MUST fail.
```

Comment:
What does fail in this context mean? Stop parsing, fail the payment, or both? 

In terms of delivering a better UX (say we want to pay, but the invoice only supports AMP and we don't), we should recommend the parser deliver a well specified error back to the caller, so this can eventually be propagated back to the user.

Reply:
Fail parsing, yes, and by implication not try to make the payment.

Reply:
We use the phrase `MUST fail the payment` elsewhere, so I've reused that:
```
   - MUST fail the payment
   - SHOULD indicate the offending bit index to the user.
```